File: two-sum-1/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    _map = {}
    nums.each_with_index do |val,i|
        n = target - val
        if _map.has_key? n
            return [i, _map[n]]
        end
        _map[val] = i
    end
end

File: two-sum-1/two_sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let _map = {};

    let i = 0;
    for(num of nums) {
        let n = target - num;
        if(n in _map) {
            return [i, _map[n]];
        }
        _map[num] = i;
        i+=1;
    }
};